### PR TITLE
gcp - add ml engine initial setup

### PR DESCRIPTION
**Pull request by mediapills**
Added initial classes with query and get functionality for locations and models resources in ML Engine service

### CODE REVIEW

File: tools/c7n_gcp/tests/test_mlengine.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        policy = self.load_policy(
+            {
+                'name': 'ml-locations-query',
+                'resource': 'gcp.ml-locations'
+            },
+            session_factory=session_factory)
+
+        resources = policy.run()
+        self.assertEqual(len(resources), 7)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_mlengine.py]: We need to be more explicit about the results we expect to be returned. What I mean by this is if we expect 7 results we should assertEqual on the specific list of 7 results. 

So if you expect a policy run to return 7 results with the names "location1, location2, location3, location4, location5, location6, location7", then verify that is what you get back in your test.

File: tools/c7n_gcp/tests/test_mlengine.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        policy = self.load_policy(
+            {
+                'name': 'ml-models-query',
+                'resource': 'gcp.ml-models'
+            },
+            session_factory=session_factory)
+
+        resources = policy.run()
+        self.assertEqual(len(resources), 1)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_mlengine.py]: We need to be more explicit about the results we expect to be returned. What I mean by this is if we expect 7 results we should assertEqual on the specific list of 7 results. 

So if you expect a policy run to return 7 results with the names "location1, location2, location3, location4, location5, location6, location7", then verify that is what you get back in your test.

**Comment by mediapills** [comment]
@bclements @egor-miasnikov the suggested changes have been done.

File: tools/c7n_gcp/c7n_gcp/resources/mlengine.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2018-2019 Capital One Services, LLC
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/mlengine.py]: just 2019

File: tools/c7n_gcp/c7n_gcp/resources/mlengine.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+@resources.register('ml-location')
+class ProjectsJobs(QueryResourceManager):
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/mlengine.py]: this class name is duplicated twice in the file and does not match the resource here. should be `MlLocation`

File: tools/c7n_gcp/c7n_gcp/resources/mlengine.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        def get(client, resource_info):
+            return client.execute_query(
+                'get', verb_arguments={
+                    'name': 'projects/{}/locations/{}'.format(
+                        resource_info['project_id'],
+                        resource_info['location'])})
+
+
+@resources.register('ml-model')
+class ProjectsJobs(QueryResourceManager):
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/mlengine.py]: should be `MlModel`

File: tools/c7n_gcp/tests/test_mlengine.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from gcp_common import BaseTest, event_data
+
+
+class TestML(BaseTest):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_mlengine.py]: need to re-run test data with class name changes I mentioned in the other file. I'm not sure why they were passing. i'm guessing they were essentially testing the same exact thing or something?

File: tools/c7n_gcp/tests/test_mlengine.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
+# Copyright 2016-2017 Capital One Services, LLC
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_mlengine.py]: should be 2019

File: tools/c7n_gcp/c7n_gcp/resources/mlengine.py (hunk 3 above)
Comment (mediapills) [reply tools/c7n_gcp/c7n_gcp/resources/mlengine.py]: @jondavidi Done

File: tools/c7n_gcp/c7n_gcp/resources/mlengine.py (hunk 4 above)
Comment (mediapills) [reply tools/c7n_gcp/c7n_gcp/resources/mlengine.py]: @jondavidi Done

File: tools/c7n_gcp/c7n_gcp/resources/mlengine.py (hunk 5 above)
Comment (mediapills) [reply tools/c7n_gcp/c7n_gcp/resources/mlengine.py]: @jondavidi Done

File: tools/c7n_gcp/tests/test_mlengine.py (hunk 6 above)
Comment (mediapills) [reply tools/c7n_gcp/tests/test_mlengine.py]: @jondavidi Done. I have been rerun tox and all works fine.

File: tools/c7n_gcp/tests/test_mlengine.py (hunk 7 above)
Comment (mediapills) [reply tools/c7n_gcp/tests/test_mlengine.py]: @jondavidi Done

**Comment by mediapills** [comment]
Hello @jondavidi,
Could you please recheck? All comments covered.
Thanks for your time!

**Comment by egor-miasnikov** [comment]
@bclements @jondavidi is this PR ready for going to `upstream`? If so please click `Approve`

**Comment by mediapills** [comment]
Created pull request into upstream [#3783](https://github.com/cloud-custodian/cloud-custodian/pull/3783).

**Comment by mediapills** [comment]
Created pull request into upstream [#3783](https://github.com/cloud-custodian/cloud-custodian/pull/3783).

**Pull request closed without merging**